### PR TITLE
Add indentation styling to SubNavLinkSection child elements

### PR DESCRIPTION
**Pull request by richorrichard**
### What does it do?

Improves visual hierarchy of sub nav panel by adjusting indentation of `<SubNavLinkSection>` child elements, and reducing overall `<SubNavLink>` top and bottom padding. 

WIP, but I believe it's complete, pending review.

### Related issue(s)/PR(s)

Related to https://github.com/strapi/design-system/issues/637

*Before*
Screen Shot 2022-06-30 at 11 21 44 AM

*After*
Screen Shot 2022-06-30 at 11 20 41 AM

### CODE REVIEW

**Comment by richorrichard** [comment]
OK - I updated the design-system to reflect the requested changes (see image for references): 
- Set spacing between `SubNavSection` title (Single Type) and its child to total 24px text-to-text using a 4px bottom margin
- Removed height limitation on `SubNavSection` title
- Removed height limitation on `SubNavLinkSection` title (Default)
- Restored previous top/bottom padding on `SubNavLink` to 8px top/bottom

Also in the image I've added additional margin below the "Click on Me" link above the Single Type `SubNavSection` title, but that's going to also have to be done in the Strapi core since that link is manually added as a button in the menu component built for the Content-Type Builder admin panel. If we like how this example looks I can commit these changes and also implement the additional margin in the Strapi core codebase afterwards.

 
Screen Shot 2022-07-13 at 4 01 09 PM

I can also discuss further on the discord - this type of adjustment is difficult to do over PR. If that's easier let me know and I can reach out there.

**Comment by gu-stav** [comment]
@richorrichard Thank you so much - this looks really good to me. What do you think?

I've added more commits:

- ae0c9592079eb07187bf53d62d0e35f7a173418e : because the min-height in the section title was removed, they collapsed if no badge was rendered. The commit makes the badge absolute positioned, so that it is included from the height computation, which allows us to set a vertical padding
- 2fe08099558d5615038f3d16eabc453ed360cf68 : renames the prop `subSectionChild` to `isSubSectionChild` to make it clear it is a boolean value

Are you ok with these changes?

From my side this PR seems good to go. I'll wait for a review from @maevalienard on Monday though before merging.

**Comment by richorrichard** [comment]
@gu-stav that all looks good to me too! I didn't realize that sometimes the badge doesn't render, so that's a good catch. 

We do need to make sure to update the core codebase to include the additional bottom padding inline for any place we use the "Add Item" button. I know that button occurs in the content-type builder; do you know anywhere else we'll need to update it?

**Comment by gu-stav** [comment]
@richorrichard Great thanks.

> We do need to make sure to update the core codebase to include the additional bottom padding inline for any place we use the "Add Item" button. I know that button occurs in the content-type builder; do you know anywhere else we'll need to update it?

I haven't worked on the content-type-builder yet, but I think we will need to add the spacing here https://github.com/strapi/strapi/blob/bf7ec7c7477fca5a6419db22dcbbd5b473e6b558/packages/core/content-type-builder/admin/src/components/ContentTypeBuilderNav/index.js#L82-L91 Currently we plan to create a new release on Monday afternoon and then either @ronronscelestes or me will take care of the design-system update in the core repository.

Would you like to finish your work there (I could give you a ping once the DS bumb PR is open) or should we do it?

**Comment by richorrichard** [comment]
I’d love to make the core update as well. Let me know when you push the DS
update and I’ll update core. Thanks!

**Comment by gu-stav** [comment]
@maevalienard I've addressed your comment in 77cbfab741a4e520eeb6790ca3dd0ac25a11d338

**Review by ronronscelestes — COMMENTED** [review]
Looks good, ready to approve on my side - I only have one question first: 
Shouldn't we do the same for v2/Subnav? 🤔 @gu-stav

**Comment by gu-stav** [comment]
@ronronscelestes Great, thanks! I do have a branch ready which ports all of these changes to v2. I haven't merged it into this one yet because it would make the review harder.

**Review by ronronscelestes — APPROVED** [review]
LGTM!

**Comment by richorrichard** [comment]
Excellent! My last question for this issue is whether or not we're ready to make the change to core to add more padding below the "+ Add Content Type" button (and similar buttons). Let me know if thats still on the table and if I should move forward. Thanks everyone!

**Comment by gu-stav** [comment]
@richorrichard We had to do a beta release before to unblock some work. I will ping you again, once the `1.2.1` release of the design-system was created, so that you can make the change in core if you want :)

**Comment by richorrichard** [comment]
Understood. I'll take a look for another issue or two to work on in the
meantime. Thanks!  





On Tue, Jul 19, 2022 at 12:31 PM Gustav Hansen ***@***.***> 
wrote:


@richorrichard  We had to do a beta release before to unblock some work. I will
ping you again, once the 1.2.1  release of the design-system was created, so
that you can make the change in core if you want :)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: 
***@***.***>

**Pull request merged**
